src/stm32/i2c.c

```diff
@@ -99,6 +107,18 @@ i2c_send_byte(I2C_TypeDef *i2c, uint8_t b, uint32_t timeout)
     i2c_wait(i2c, I2C_SR1_TXE, 0, timeout);
 }
 
+static void
+i2c_read_byte(I2C_TypeDef *i2c, uint8_t* read, uint32_t timeout,
+              uint8_t remaining)
+{
+    i2c_wait(i2c, I2C_SR1_RXNE, 0, timeout);
+    irq_disable();
+    *read = i2c->DR;
```

Comment:
Very minor, comment - gcc often does a poor job generating code for pass-by-reference parameters.  Might be better to change this function to return a uint8_t.